types/node/index.d.ts

```diff
@@ -794,6 +794,7 @@ declare module "http" {
         timeout: number;
         listening: boolean;
     }
+    export class Server {}
```

Comment:
A Http server has a constructor with optional parameter request listener. Additionally it extends `net.Server`. I think the correct approach here would be to replace the interface server by a class Server instead of exporting both.
see also #17720 where this (and more) was proposed.